ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -369,7 +369,17 @@ - (void)promptTouchId {
                                                             }]];
                     [strongSelf presentViewController:alert animated:YES completion:nil];
                 } else if (error.code == LAErrorUserCancel) {
-                    [strongSelf makePasscodeViewBecomeFirstResponder];
+
+                    // call becomeFirstResponder here to show the keyboard. dispatch to main queue with
```

Comment:
Is the delay really necessary? From the documentation on evaluatePolicy:localizedReason:reply:, the reply block (i.e. this code) "[...] is evaluated on a private queue internal to the framework in an unspecified threading context", so I would expect odd behavior if you tried to set the first responder from here (UIKit not being thread-safe and all). Does dispatching async to the main queue without a delay not suffice?